src/utils/getFetchPolicyForKey.ts

```diff
@@ -0,0 +1,16 @@
+import { WatchQueryFetchPolicy } from '@apollo/client'
+
+const keys = new Map<string, number>()
+
```

Comment:
this implements a time based cache policy. if a key doesn't exist it stores it at the timestamp and the next time it queries it it checks if its within the expiration time. if it is it sets the fetch-policy to use the cache if its outside it sets it to use the network